rfd/0027-account-lifecycle.md

```diff
@@ -0,0 +1,195 @@
+the resources.
+
+The possible vectors of attack is for malicious actor to overtake someone's
+account with a request to delete all the data. The grace period and active privileged
+user should mitigate against that.
+
+**External request for canceled account**
+
+If the user used their @gmail address, they won't be able to recover their account with us,
+but can request cancellation.
+
+* Request account cancellation by clicking on "cancel my account link" URL.
+* The user should have write privileges on the billing information for the cancellation to work.
+* We email crypto random 32 byte hex cancel account link to their email address.
+* They click the link and we mark the account as "inactive".
+
+```
+We have marked your account as inactive. In 10 days, we are going to delete the account and all it's data.
+To prevent this, log into Teleport and press "revert account cancelation".
+```
+
+If anyone logs into this account, we should show them the warning:
+
+```
+Your account was marked as inactive at request of `email`.
+If you would like to revert acount from being canceled in x days, go to billing page
+and press "revert account cancelation".
+```
+
+All users with write billing permission in the email account should get the following notification over email:
+
+```
+Your account was marked as inactive at request of `email`.
+The account and all data will be deleted after 10 days grace period.
+To prevent this, login using your credentials and press "revert account cancelation".
+```
+
+Only users with write permissions to billing should be able to revert account
+cancelation. We should asking a user for second factor proof to revert account cancelation to prevent
+a possible account overtake by a malicious actor who wishes to keep the account for accessing
+the resources.
+
+The possible vectors of attack is for malicious actor to overtake someone's
+email account and launch an attack to delete the data. The grace period and the broadcast
+notification should allow users to mitigate the deletion by logging in their account.
+
+We should recommend users to have at least two users with write billing permissions
+so the other person will be notified and can act.
+
+
+**When all is lost**
+
+In case if a system administrator has lost everything, we can not automate the recovery flow.
+For social emails @gmail.com and so on, we can't have any connection between a user
+and their account, so they only will be able to request cancel of their account using
+external flow.
+
+
+If it's an organization whose users have @corp domain, we can ask for advanced slow validation
+using offline methods:
+
+* The company acme.com should proove the ownership of their domain
+by posting their registered U.S. corporate mail address in the TEXT record:
+
+teleport.acme.com IN TXT <corp address, signing officer, verifying officer>
+
+Our support should verify that this address matches their registered office location
```

Comment:
I'm not sure what this does, sounds like a verification against public record, as long as the corp is public?